nautilus_trader/data/aggregation.pyx

```diff
@@ -765,90 +821,77 @@ cdef class TimeBarAggregator(BarAggregator):
             The timestamp (UTC).
 
         """
-        step = self.bar_type.spec.step
-        aggregation = self.bar_type.spec.aggregation
+        cdef int step = self.bar_type.spec.step
```

Comment:
Good catch here on variable type declaration.

Sometimes it's not necessary to declare C types for local variables if it won't affect performance so much, but it _can_ help with correctness.